waku/waku_discv5.nim

```diff
@@ -78,33 +79,24 @@ proc new*(
   peerManager: Option[PeerManager] = none(PeerManager),
   queue: AsyncEventQueue[SubscriptionEvent] = newAsyncEventQueue[SubscriptionEvent](30),
   ): T =
-  let shardPredOp =
-    if record.isSome(): shardingPredicate(record.get())
-    else: none(WakuDiscv5Predicate)
-  
-  var bootstrapRecords = conf.bootstrapRecords
-
-  # Remove bootstrap nodes with which we don't share shards.
-  if shardPredOp.isSome():
-    bootstrapRecords.keepIf(shardPredOp.get())
-  
-  if conf.bootstrapRecords.len > 0 and bootstrapRecords.len == 0:
-    warn "No discv5 bootstrap nodes share this node configured shards"
-
   let protocol = newProtocol(
     rng = rng,
     config = conf.discv5Config.get(protocol.defaultDiscoveryConfig),
     bindPort = conf.port,
     bindIp = conf.address,
     privKey = conf.privateKey,
-    bootstrapRecords = bootstrapRecords,
+    bootstrapRecords = conf.bootstrapRecords,
     enrAutoUpdate = conf.autoupdateRecord,
     previousRecord = record,
     enrIp = none(ValidIpAddress),
     enrTcpPort = none(Port),
     enrUdpPort = none(Port),
   )
 
+  let shardPredOp =
+    if record.isSome(): shardingPredicate(record.get(), conf.bootstrapRecords)
+    else: none(WakuDiscv5Predicate)
```

Comment:
Not a big deal at all but maybe we can simplify a little by setting a default argument to the `new` proc:
Something like:
```
record: Option[waku_enr.Record] = none(WakuDiscv5Predicate)
```

Reply:
The types would be different no?